server/src/main/java/org/elasticsearch/search/aggregations/bucket/geogrid/BoundedGeoTileGridTiler.java

```diff
@@ -0,0 +1,107 @@
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.elasticsearch.search.aggregations.bucket.geogrid;
+
+import org.elasticsearch.common.geo.GeoBoundingBox;
+import org.elasticsearch.common.geo.GeoRelation;
+import org.elasticsearch.geometry.Rectangle;
+import org.elasticsearch.index.fielddata.MultiGeoValues;
+
+public class BoundedGeoTileGridTiler extends GeoTileGridTiler {
+    private final double boundsTop;
+    private final double boundsBottom;
+    private final double boundsWestLeft;
+    private final double boundsWestRight;
+    private final double boundsEastLeft;
+    private final double boundsEastRight;
+    private final boolean crossesDateline;
+
+    public BoundedGeoTileGridTiler(GeoBoundingBox geoBoundingBox) {
+        // split geoBoundingBox into west and east boxes
+        boundsTop = geoBoundingBox.top();
+        boundsBottom = geoBoundingBox.bottom();
+        if (geoBoundingBox.right() < geoBoundingBox.left()) {
+            boundsWestLeft = -180;
+            boundsWestRight = geoBoundingBox.right();
+            boundsEastLeft = geoBoundingBox.left();
+            boundsEastRight = 180;
+            crossesDateline = true;
+        } else { // only set east bounds
+            boundsEastLeft = geoBoundingBox.left();
+            boundsEastRight = geoBoundingBox.right();
+            boundsWestLeft = 0;
+            boundsWestRight = 0;
+            crossesDateline = false;
+        }
+    }
+
+    public int advancePointValue(long[] values, double x, double y, int precision, int valuesIdx) {
+        long hash = encode(x, y, precision);
+        if (cellIntersectsGeoBoundingBox(GeoTileUtils.toBoundingBox(hash))) {
```

Comment:
I think in the case of a point, we can just check that the point is in bounds? We save having to compute the rectangle.

Reply:
I believe we discussed this and decided we would like to keep a consistent definition between points and shapes where we only filter which cells we consider with regards to the geoBoundingBox. this means the point need not be within the geoBoundingBox

Reply:
yes, yes, my bad :(

Reply:
It is exactly how we spoke and I think it should be

Reply:
👍